yesod-core/src/Yesod/Core/Internal/Run.hs

```diff
@@ -116,7 +117,7 @@ basicRunHandler rhe handler yreq resState = do
             return (HCContent defaultStatus tc))
         (\e ->
             case fromException e of
-                Just e' -> return e'
+                Just (AnnotatedException _ e') -> return e'
```

Comment:
This deserves a bit of commentary.

The [`instance Exception e => Exception (AnnotatedException e)`](https://hackage.haskell.org/package/annotated-exception-0.2.0.0/docs/Control-Exception-Annotated.html#t:AnnotatedException) has a slightly ~magical~ ~terrible~ ~horrifying~ unusual implementation of `fromException`.

The [tests for this demonstrate the pattern](https://github.com/parsonsmatt/annotated-exception/blob/master/test/Control/Exception/AnnotatedSpec.hs#L41-L65), but tl;dr, the `Exception` instance here on `AnnotatedException` will work to catch both an `AnnotatedException HandlerContents` *and* a `HandlerContents`, which is then promoted to `AnnotatedException mempty (hc :: HandlerContents)`.